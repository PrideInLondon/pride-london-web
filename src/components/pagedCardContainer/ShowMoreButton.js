import React from 'react'
import PropTypes from 'prop-types'
import { Column, Row } from '../grid'
import { Button } from '../button'

const ShowMoreButton = ({ text, onClick }) => (
  <Row pb={[30, 30, 50]}>
    <Column mx="auto" pt={[30, 30, 50]}>
      <Button onClick={onClick}>{text}</Button>
    </Column>
  </Row>
)

ShowMoreButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default ShowMoreButton
