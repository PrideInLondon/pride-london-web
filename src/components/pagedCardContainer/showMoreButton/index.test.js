import React from 'react'
import { shallow } from 'enzyme'
import ShowMoreButton from '.'

describe('ShowMoreButton', () => {
  const text = 'Foo'
  const onClick = jest.fn()

  let wrapper
  beforeAll(() => {
    wrapper = shallow(<ShowMoreButton text={text} onClick={onClick} />)
  })

  it('calls the onClick when clicked', () => {
    expect(onClick).not.toHaveBeenCalled()

    wrapper.find('Button').simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
