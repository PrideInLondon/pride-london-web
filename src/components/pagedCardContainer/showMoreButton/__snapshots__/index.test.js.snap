// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreButton renders component 1`] = `
<Gridstyles__Row
  display="flex"
  flexWrap="wrap"
  pb={
    Array [
      30,
      30,
      50,
    ]
  }
>
  <Gridstyles__Column
    mx="auto"
    pt={
      Array [
        30,
        30,
        50,
      ]
    }
  >
    <Button
      className=""
      disabled={false}
      flexwidth={false}
      fullmobile={false}
      link={false}
      onClick={[MockFunction]}
      primary={true}
      small={false}
      to={null}
      type="button"
      white={false}
      wide={true}
    >
      Foo
    </Button>
  </Gridstyles__Column>
</Gridstyles__Row>
`;
